File: atcoder/atcoder_problems/arc001~100/arc086/arc086_a.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int N, K; 
    cin >> N >> K;
    vector<int> A(N, 0);
    for (int _ = 0; _ < N; _++){
        int a;
        cin >> a;
        A[a - 1] += 1;
    }

    sort(A.begin(), A.end());
    
    int ans = 0;
    for (int i = 0; i < A.size() - K; i++){
        ans += A[i];
    }
    cout << ans << endl;
    return 0;
}





// #include <iostream>
// #include <vector>
// #include <algorithm>
// using namespace std;

// int main(){
//     int N, K; 
//     cin >> N >> K;
//     vector<int> A(N, 0);
//     for (int _ = 0; _ < N; _++){
//         int a;
//         cin >> a;
//         A[a - 1] += 1;
//     }

//     sort(A.begin(), A.end(), greater<int>());
    
//     int ans = 0;
//     for (int i = K; i < A.size(); i++){
//         ans += A[i];
//     }
//     cout << ans << endl;
//     return 0;
// }

File: atcoder/atcoder_problems/arc001~100/arc086/arc086_a.py
N, K = map(int, input().split())
A = list(map(int, input().split()))

A_count_list = [0 for _ in range(200001)]

for i in range(N):
    A_count_list[A[i] - 1] += 1

A_count_list.sort(reverse=True)

ans = 0
for i in range(K,200001):
    if A_count_list[i] == 0:
        break
    ans += A_count_list[i]


print(ans)


# N, K = map(int, input().split())
# A = list(map(int, input().split()))

# ball = [0] * N

# for i in range(N):
#     ball[A[i]-1] += 1

# ball.sort(reverse=True)
# ans = sum(ball[K:])
# print(ans)